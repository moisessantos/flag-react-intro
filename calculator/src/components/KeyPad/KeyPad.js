import React, { memo } from 'react';

import { KeyRow } from '../';

const KeyPad = ({ ...otherProps }) => (
  <div className="button" {...otherProps}>
    <KeyRow values={['(', 'CE', ')', 'C']} />
    <KeyRow values={['1', '2', '3', '+']} />
    <KeyRow values={['4', '5', '6', '-']} />
    <KeyRow values={['7', '8', '9', '*']} />
    <KeyRow values={['.', '0', '=', '/']} />
  </div>
);

export default memo(KeyPad);
