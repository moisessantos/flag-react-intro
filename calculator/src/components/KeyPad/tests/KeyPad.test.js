import React from 'react';
import { shallow } from 'enzyme';
import KeyPad from '../';

describe('KeyPad tests', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<KeyPad />)
        expect(wrapper).toMatchSnapshot();
    });
})