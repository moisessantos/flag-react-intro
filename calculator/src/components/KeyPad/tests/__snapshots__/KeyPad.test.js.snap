// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPad tests should render as expected 1`] = `
<div
  className="button"
>
  <Memo(KeyRow)
    values={
      Array [
        "(",
        "CE",
        ")",
        "C",
      ]
    }
  />
  <Memo(KeyRow)
    values={
      Array [
        "1",
        "2",
        "3",
        "+",
      ]
    }
  />
  <Memo(KeyRow)
    values={
      Array [
        "4",
        "5",
        "6",
        "-",
      ]
    }
  />
  <Memo(KeyRow)
    values={
      Array [
        "7",
        "8",
        "9",
        "*",
      ]
    }
  />
  <Memo(KeyRow)
    values={
      Array [
        ".",
        "0",
        "=",
        "/",
      ]
    }
  />
</div>
`;
